subsys/settings/Kconfig

```diff
@@ -45,7 +45,7 @@ config SETTINGS_USE_BASE64
 
 choice
 	prompt "Storage back-end"
-	default SETTINGS_FCB if FCB
```

Comment:
Should we have both lines perhaps? It would make sense to have match settings backend with flash format.

Reply:
Does Kconfig work like that?, do default assignment have severity depends on they are placed higher or lower?

Reply:
First match I think.

Reply:
Example here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/host/Kconfig#L28

Reply:
@kapi-no Yes, Indeed:
> - default value: "default" <expr> ["if" <expr>]
  A config option can have any number of default values. If multiple
  default values are visible, only the first defined one is active.

https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt

Reply:
@kapi-no 
Either we list defaults, or we don't
So either:
```
choice
	prompt "Storage back-end"
	default SETTINGS_NVS if NVS
	default SETTINGS_FCB if FCB
	default SETTINGS_FS if FILE_SYSTEM
	default SETTINGS_NONE
	depends on SETTINGS
```
or
```
choice
	prompt "Storage back-end"
	depends on SETTINGS
```
Which would result in the first one (I think).

Reply:
I decided to list the defaults, the configuration for samples is a bit simpler when defaults are there